libcoap/src/oscore.rs

```diff
@@ -0,0 +1,142 @@
+use core::{ffi::c_void, ptr};
```

Comment:
License/Copyright header is missing.
```suggestion
// SPDX-License-Identifier: BSD-2-Clause
/*
 * Copyright © The libcoap-rs Contributors, all rights reserved.
 * This file is part of the libcoap-rs project, see the README file for
 * general information on this project and the NOTICE.md and LICENSE files
 * for information regarding copyright ownership and terms of use.
 *
 * oscore.rs - Wrapper for libcoap OSCORE functionality.
 */
 
use core::{ffi::c_void, ptr};
```

Additionally, some module-level documentation describing the general method of using OSCORE (and possibly examples) would be nice here, maybe you could adapt the documentation from the Markdown you sent me.